sw/device/tests/sim_dv/keymgr_sideload_kmac_test.c

```diff
@@ -0,0 +1,154 @@
+  CHECK_DIF_OK(
+      dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  // Configure KMAC hardware (using software key and software entropy).
+  kmac_testutils_config(&kmac, false);
+
+  uint32_t output[kKmacOutputLen];
+  kmac_testutils_kmac(&kmac, kKmacMode, &kSoftwareKey, kCustomString,
+                      kCustomStringLen, kKmacMessage, kKmacMessageLen,
+                      kKmacOutputLen, output);
+  LOG_INFO("Computed KMAC output for software key.");
+
+  // Check that the output matches the expected output.
+  CHECK_ARRAYS_EQ(output, kSoftwareKeyExpectedOutput, kKmacOutputLen);
+
+  // Reconfigure KMAC to use the sideloaded key.
+  kmac_testutils_config(&kmac, true);
+
+  kmac_testutils_kmac(&kmac, kKmacMode, &kSoftwareKey, kCustomString,
+                      kCustomStringLen, kKmacMessage, kKmacMessageLen,
+                      kKmacOutputLen, output);
+  LOG_INFO("Computed KMAC output for sideloaded key.");
+
+  // Check that the output is different from the software key output
+  // (indicating the software key was not used).
+  //
+  // TODO(weicai): also check in SV sequence that the KMAC output is correct.
+  CHECK_ARRAYS_NE(output, kSoftwareKeyExpectedOutput, kKmacOutputLen);
+}
+
+bool test_main(void) {
+  dif_rstmgr_t rstmgr;
+  dif_rstmgr_reset_info_bitfield_t info;
+
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+  info = rstmgr_testutils_reason_get();
+
+  // POR reset.
+  if (info == kDifRstmgrResetInfoPor) {
+    LOG_INFO("Powered up for the first time, program flash");
+
+    keymgr_testutils_init_flash();
+
+    // Lock otp secret partition.
+    dif_otp_ctrl_t otp;
+    CHECK_DIF_OK(dif_otp_ctrl_init(
+        mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR), &otp));
+    otp_ctrl_testutils_lock_partition(&otp, kDifOtpCtrlPartitionSecret2, 0);
+
+    // Reboot device.
+    rstmgr_testutils_reason_clear();
+    CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
+
+    // Wait here until device reset.
+    wait_for_interrupt();
+
+  } else if (info == kDifRstmgrResetInfoSw) {
+    LOG_INFO(
+        "Powered up for the second time, actuate keymgr and perform test.");
+
+    // Initialize keymgr.
+    dif_keymgr_t keymgr;
+    mmio_region_t base_addr =
+        mmio_region_from_addr(TOP_EARLGREY_KEYMGR_BASE_ADDR);
+    CHECK_DIF_OK(dif_keymgr_init(base_addr, &keymgr));
+
+    keymgr_testutils_check_state(&keymgr, kDifKeymgrStateReset);
+    keymgr_testutils_advance_state(&keymgr, NULL);
+    keymgr_testutils_check_state(&keymgr, kDifKeymgrStateInitialized);
+    LOG_INFO("Keymgr entered Init State");
+
+    // Advance to CreatorRootKey state.
+    keymgr_testutils_advance_state(&keymgr, &kCreatorParams);
+    keymgr_testutils_check_state(&keymgr, kDifKeymgrStateCreatorRootKey);
+    LOG_INFO("Keymgr entered CreatorRootKey State");
+
+    // Test KMAC sideloading.
+    test_kmac_with_sideloaded_key();
```

Comment:
I may miss something. I didn't find where we generate a sideload key for kmac. We may need to call this function with dest - kDifKeymgrVersionedKeyDestAes.
`dif_keymgr_generate_versioned_key();`

same for the aes sideload test.